### ISSUE TITLE
[Discovery] Can we detect presence/absence of IAM session reliably?

### ISSUE THREAD

**Issue by patrickvinograd** [department-of-veterans-affairs/va.gov-team#4651]
Can we reliably get a yes/no on whether an IAM session exists and is valid? Potentially be examining the session cookie?

Assuming we can:
If session is present and we can resolve the asymmetric UUID issue, then we can hopefully use an IAM session to boostrap a VA.gov session.

If session is not present, then we may want to use that to log out a VA.gov session, similar to how the shared MHV session cookie is implemented today.  (Though this part is subject to the complexities of differing session durations).

**Comment by ericbuckley** [comment]
I believe the session cookie that IAM is setting is on the subdomain, for example `pint.eauth.va.gov` therefore we won't be able to read it from within the vets-api.  However if they we're to set the session cookie on `.va.gov` we potentially could.

I'm not sure how many different applications run within the `va.gov` domain space though, so if there are a lot I could see some issues with security here that may prevent the IAM team from wanting to do that.

@patrickvinograd are there any documents out there that list the subdomains within `va.gov`?

**Comment by ambikaroos** [comment]
@ericbuckley to ask IAM whether they have a best practice recommendation for knowing if a session is established, maybe they set an indicator cookie. The cookie they set right now, we aren't able to read.

**Comment by patrickvinograd** [comment]
Wondering  if we can work with IAM to establish a domain-scoped cookie (like described here: https://stackoverflow.com/questions/18492576/share-cookie-between-subdomain-and-domain) that would be usable across *.va.gov domains. This  could potentially not be the canonical IAM session cookie, but an "indicator" cookie that all sites could read  to know if an IAM session was present and use as a basis for UX decisions.

**Comment by patrickvinograd** [comment]
Alternatively, if we  can use the same trick as the session keepalive, requesting a favicon or other protected  resource, and seeing if it succeeds or fails...don't  know if  we can do that in a browser  background worker without violating  same-domain restrictions?

**Comment by ambikaroos** [comment]
Per conversation with Damien on today's cross-team call, waiting to hear back from Craig hopefully early next week

**Comment by ericbuckley** [comment]
We can test for a active vs inactive session from the frontend by pinging the https://pint.eauth.va.gov/keepalive. A response with a content-type: text/html means the session is inactive, however a response with a content-type: text/plain or application/json means the session is active.

**Comment by ericbuckley** [comment]
tldr; the https://pint.eauth.va.gov/keepalive endpoint doesn't work outside of the PINT domain, the CORS policy needs to be updated.

The tests I ran last week worked, because all of them we're initiated from the same origin (meaing the call to the url came from the PINT domain).  However for us to use this we will need to run an XHR request from a different domain, in production this will be https://www.va.gov/.

The IAM team has set the CORS header, `access-control-allow-origin: *`, so in theory the request should work.  However in our case we need to send the users cookie over, otherwise all requests will result in an inactive user session.

However if we want to send the users cookie over using the [withCredentials flag](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials), the browser will only do so if the [access-control-allow-origin header](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSNotSupportingCredentials) is explicitly set.

I'm going to reach out to the IAM team now to see if they can make some updates to explicitly set this value for us.  @hinzed1127 thanks for helping me test this out.

**Comment by ericbuckley** [comment]
I went back and forth with Marat on the IAM team on this today and he was finally able to make a change that updated the CORS header to give us access to do this.

I ran a test this evening and it worked for me, below are the steps I ran to confirmed that it worked.  @hinzed1127 when you have a chance, could you also run a quick test, maybe using a JS fetch call in the login modal?  If its working for both of us, then I think we can close this spike and open an implementation ticket.

```
- open up a private tab in Firefox (also tested with Chrome)
- go to https://pint.eauth.va.gov/accessva/
- click on the Vendor Information Pages image, and start a ID.ME login
- log into ID.ME with user vets.gov.user+262@gmail.com
- open up a new tab to confirm that https://pint.eauth.va.gov/keepalive returns {200, "OK"}
- open up a new tab for https://www.va.gov/, with the devtools console open
- run the following script to test an XHR request
  var req = new XMLHttpRequest();
  req.open('GET', 'https://pint.eauth.va.gov/keepalive', false);
  req.setRequestHeader('Content-Type', 'application/json');
  req.withCredentials = true;
  req.send();
```

NOTE: the payload for the `/keepalive` endpoint was updated to return `{"status": 200, "message": "OK"}` for active sessions rather than the previous value of `{200,"OK"}` that we were seeing.

**Comment by ericbuckley** [comment]
@ambikaroos I beleive we can close this in favor of #5417 and #5737
